### PR TITLE
Helios Server

### PR DESCRIPTION
**Pull request by begizi**

### CODE REVIEW

**Comment by begizi** [comment]
@samccone How should the travisci file be set up now that it isn't only a node app?

**Comment by samccone** [comment]
you can change the type of the app in the travis.yml

File: api/src/helios/weather/weather.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func initWeatherFetch(apiURL string) {
+	for {
+		weather := getWeather(apiURL)
+
+		WeatherChan <- weather
+
+		// Timeout for 60 seconds before next request
+		time.Sleep(60 * time.Second)
```
Comment (joelhawksley) [inline api/src/helios/weather/weather.go]: This will max out our daily API limit of 1000 requests. How about once every 120 seconds?

Comment (samccone) [inline api/src/helios/weather/weather.go]: Or every 10 min 😎🐴🏪🌇🌋

File: api/src/helios/weather/weather.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	panic("Shouldn't be here")
+}
+
+func getWeather(apiURL string) interface{} {
+	var weatherBody interface{}
+
+	resp, err := http.Get(apiURL)
+	if err != nil {
+		log.Warn("Failed to fetch weather", "error", err.Error())
```
Comment (joelhawksley) [inline api/src/helios/weather/weather.go]: Optional: Might we consider returning an error message to the client?

File: client/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Helios Client
+
+### Developing
+
+1. `npm install`
+2. Create `config.json` with API_URL containing Forecast.io API key
```
Comment (joelhawksley) [inline client/README.md]: I assume we don't need this any more?

**Comment by joelhawksley** [comment]
@begizi do you think we should add the socket connection for the weather data in this PR, or do it in a future PR?

**Comment by begizi** [comment]
@joelhawksley I'd probably do it in a future PR. We need to implement a cache on the server so when a web socket connects we can send all of the latest service payloads so there isn't a 0-120 second delay before weather data shows up.

File: client/README.md (hunk 3 above)
Comment (joelhawksley) [inline client/README.md]: Resolved.

File: api/src/helios/weather/weather.go (hunk 2 above)
Comment (begizi) [inline api/src/helios/weather/weather.go]: That would probably be useful for being able to visually represent when a service is down on the client. Let's make this a separate ticket though. We'll need to wrap our response with some kind of status and error message attribute.

**Comment by begizi** [comment]
@joelhawksley Updated.

File: api/src/helios/weather/weather.go (hunk 2 above)
Comment (joelhawksley) [inline api/src/helios/weather/weather.go]: https://github.com/mojotech/helios/issues/39

File: api/src/helios/weather/weather.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	}
+}
+
+func initWeatherFetch(apiURL string) {
+	for {
+		weather := getWeather(apiURL)
+
+		WeatherChan <- weather
+
+		// Timeout for 60 seconds before next request
```
Comment (joelhawksley) [inline api/src/helios/weather/weather.go]: :flushed:

**Comment by joelhawksley** [comment]
merge when green

File: api/src/helios/weather/weather.go (hunk 4 above)
Comment (begizi) [inline api/src/helios/weather/weather.go]: :cry: :panda_face:

**Comment by samccone** [comment]
we should make sure we can deploy before merge :)

**Comment by joelhawksley** [comment]
@samccone good point. @begizi want to add a release commit on top of this, so we can test the whole stack?

**Comment by begizi** [comment]
I tested the release files on my pi last night and everything worked. We'll have the change the deploy script for sure though

**Pull request merged**